pandas/core/indexes/multi.py

```diff
@@ -2676,10 +2676,17 @@ def _partial_tup_index(self, tup, side="left"):
                 return start + section.searchsorted(loc, side=side)
 
             idx = self._get_loc_single_level_index(lev, lab)
-            if k < n - 1:
+            if isinstance(idx, slice) and k < n - 1:
+                # Get start and end value from slice, necessary when a non-integer
+                # interval is given as input GH#37707
+                start = idx.start
+                end = idx.stop
+            elif k < n - 1:
                 end = start + section.searchsorted(idx, side="right")
                 start = start + section.searchsorted(idx, side="left")
             else:
+                if isinstance(idx, slice):
```

Comment:
can you make this an if/else and then an else case

Reply:
This way?

Reply:
umm sort of, L2688 should be an elif on L2687

Reply:
Sorry, should be ok now